src/hdrgen.d

```diff
@@ -1742,10 +1742,16 @@ public:
         //printf("FuncDeclaration::toCBuffer() '%s'\n", f.toChars());
         if (stcToBuffer(buf, f.storage_class))
             buf.writeByte(' ');
-        typeToBuffer(f.type, f.ident);
+        auto tf = cast(TypeFunction)f.type;
+        typeToBuffer(tf, f.ident);
         if (hgs.hdrgen == 1)
         {
-            if (f.storage_class & STCauto)
+            if(f.fbody && f.fbody.toChars()[0] == '\0')
```

Comment:
As I see it, the only case where the 2 are not mutually exclusive is when you have an empty function which is auto. I will take this into consideration